overlay.js

```diff
@@ -0,0 +1,333 @@
+document.addEventListener('keydown', documentEvent => {
+    if (documentEvent.shiftKey && documentEvent.key === 'L' && !document.querySelector('#skater-overlay')) {
+        // const browser = chrome || browser;
+        createOverlay();
+        setUpInputEventListener();
+        setTimeout(() => focusInput(), 100);
+    } else if (getSearchInputElement()) {
+        focusInput();
+        const focusedElement = document.querySelector('.focused');
+        switch(documentEvent.key) {
+            case "Up":
+            case "ArrowUp":
+                // move to last search result or input
+                // FIXME functionize
+                if (focusedElement.getAttribute('class') === 'skater-link skater-result-0 focused') {
+                    focusInput();
+                    // FIXME: implement preventDefault better
+                    documentEvent.preventDefault();
+                    return true
+                } else if (focusedElement.isSameNode(getSearchInputElement())) {
+                    return true
+                } else {
+                    // move to previous result
+                    moveUpOneResult();
+                    documentEvent.preventDefault();
+                    return true
+                }
+            case "Down":
+            case "ArrowDown":
+                    // move to next search result
+                    // FIXME: check if this is the last element
+                    documentEvent.preventDefault();
+                    moveDownOneResult();
+                    return true
+            case "Enter":
+                // go to first inputEvent in the list
+                const selectedResult = document.querySelector('.focused');
+                console.log(selectedResult);
+                // TODO use chrome.tabs.create by sending this as message to background.js
+                if (documentEvent.ctrlKey){
+                    // open in same window
+                } else {
+                    console.log(selectedResult);
+                    window.open(selectedResult.href);
+                    destroyOverlay();
+            }
+        }
+    }
+    if (documentEvent.key === "Escape") {
```

Comment:
this has similar logic to what your switch is but is doing so seperately. probably want to combine similar logic